cloud/amazon/rds_snapshot_copy.py

```diff
@@ -0,0 +1,155 @@
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: rds_snapshot_copy
+short_description: copies an existing rds snapshot
+description:
+    - copies an existing rds snapshot
+    - source snapshot must be in available state
+options:
+  source_region:
+    description:
+      - source region
+      - not provided then the environment default will be assumed
+    required: false
+  source_db_snapshot_identifier:
+    description:
+      - identifier of the source db snapshot
+      - must be in the "available" state
+      - if the source is in the same region then specify the snapshot id
+      - if the source is in a different region then specify the snapshot ARN
+    required: true
+  target_db_snapshot_identifier:
+    description:
+      - identifier for the copied snapshot
+      - cannot be empty
+      - must contain 1-255 alphanumeric characters or hyphens
+      - first character must be a letter
+      - cannot end with a hypen or contain two consecutive hyphens
+    required: true
+
+author: "Brock Haywood (@brockhaywood)"
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+
+'''
+
+try:
+    import boto.rds
+    HAS_BOTO = True
+except ImportError:
+    HAS_BOTO = False
+
+
+def copy_snapshot(module, source_conn, conn,
+                  source_db_snapshot_identifier=None,
+                  target_db_snapshot_identifier=None):
+
+    if not source_db_snapshot_identifier or not target_db_snapshot_identifier:
+        module.fail_json(msg='Source DB Snapshot Id and Target DB Snapshot '
+                             'Id must be provided')
+
+    try:
+        snapshots = conn.get_all_dbsnapshots(
+            snapshot_id=source_db_snapshot_identifier)
+    except boto.exception.BotoServerError, e:
+        module.fail_json(msg = "%s: %s" % (e.error_code, e.error_message))
+
+    if not snapshots:
+        module.fail_json(msg="Could not find snapshot with "
+                             "identifier %s" % (source_db_snapshot_identifier))
+
+    snapshot = snapshots[0]
+
+    if snapshot.status != 'available':
+        module.fail_json(msg="Snapshot not available with identifier "
+                             "%s" % (source_db_snapshot_identifier))
+
+    snapshot = conn.copy_dbsnapshot(
+        source_snapshot_id=source_db_snapshot_identifier,
+        target_snapshot_id=target_db_snapshot_identifier
+    )
+
+    module.exit_json(changed=True,
+                     snapshot_id=snapshot.id)
+
+
+def copy_rds_snapshot_ansible_module():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            region = dict(),
+            source_region = dict(),
+            source_db_snapshot_identifier = dict(),
+            target_db_snapshot_identifier = dict(),
+        )
+    )
+    module = AnsibleModule(argument_spec=argument_spec)
+    return module
+
+
+def main():
+    module = copy_rds_snapshot_ansible_module()
```

Comment:
Hmm, that's not how I originally wrote `ec2_snapshot`. Apparently the changes make it "more testable".

Either way, not your fault, and this is not a blocker.